### PR TITLE
Fixes #22846: Update user management to manage secondary group of user

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/22846

### CODE REVIEW

File: techniques/systemSettings/userManagement/userManagement/9.1/metadata.xml [hunk 1]
```diff
@@ -133,6 +133,30 @@
           <DEFAULT>false</DEFAULT>
         </CONSTRAINT>
       </SELECT1>
+      <INPUT>
+        <NAME>USERGROUP_USER_SECONDARY_GROUPS</NAME>
+        <DESCRIPTION>Secondary groups name for this user, comma separated)</DESCRIPTION>
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/9.1/metadata.xml:138]: Unecessary parenthesis at the end

File: techniques/systemSettings/userManagement/userManagement/9.1/metadata.xml [hunk 2]
```diff
@@ -133,6 +133,30 @@
...
+        <LONGDESCRIPTION>On UNIX systems, ensure that the user belongs to the list of groups, as secondary groups</LONGDESCRIPTION>
+        <CONSTRAINT>
+          <MAYBEEMPTY>true</MAYBEEMPTY>
+        </CONSTRAINT>
+      </INPUT>
+      <SELECT1>
+        <NAME>USERGROUP_FORCE_USER_SECONDARY_GROUPS</NAME>
+        <DESCRIPTION>Enforce the secondary groups of the user</DESCRIPTION>
+        <LONGDESCRIPTION>If set to exactly, the user will belong exactly to the list of secondary groups, otherwise, it may also be in other groups</LONGDESCRIPTION>
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/9.1/metadata.xml:147]: I would have named that "exclusive" instead.

File: techniques/systemSettings/userManagement/userManagement/9.1/userManagement.st [hunk 3]
```diff
@@ -104,6 +110,11 @@ bundle agent check_usergroup_user_parameters_&RudderUniqueID&
         string => "-u ${usergroup_user_uid[${usergroup_user_index}]}",
             if => "!usermanagement_user_uid_empty_${usergroup_user_index}";
 
+      # define inner class prefix of secondary group management
+      "inner_classprefix_secondary_group_${usergroup_user_index}"
+        string => canonify("user_secondary_groups_${usergroup_user_login[${usergroup_user_index}]}"),
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/9.1/userManagement.st]: This does not look like a class prefix but an old_class_prefix, is that really what we want to use?

File: techniques/systemSettings/userManagement/userManagement/9.1/userManagement.st [hunk 4]
```diff
@@ -392,6 +407,40 @@ bundle agent check_usergroup_user_parameters_&RudderUniqueID&
   methods:
     pass3.showtime::
 
+      ######################
+      # set/check secondary groups
+      # if checkpres, then we are doing dry run only
+      ######################
+      "force_dry_run_mode_${usergroup_user_login[${usergroup_user_index}]}_${usergroup_user_secondary_groupsname[${usergroup_user_index}]}"
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/9.1/userManagement.st:415]: I am pretty sure only the login is needed in the promiser here.

Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/9.1/userManagement.st:415]: For the 3 promisers below, I am pretty sure we only need to put the logins there

Comment (ncharles) [reply techniques/systemSettings/userManagement/userManagement/9.1/userManagement.st:415]: no, if we manage the same user in different way, we need both.
index will even be worse if we have 2 directives based on the same technique

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by Fdall — APPROVED** [review]

**Comment by ncharles** [comment]
PR updated with a new commit

**Pull request merged**